litgpt/lora.py

```diff
@@ -512,7 +512,7 @@ def __init__(self, config: Config) -> None:
         self.transformer = nn.ModuleDict(
             dict(
                 wte=nn.Embedding(config.padded_vocab_size, config.n_embd),
-                h=nn.ModuleList(Block(config) for _ in range(config.n_layer)),
+                h=nn.ModuleList(Block(config, block_idx) for block_idx in range(config.n_layer)),
```

Comment:
yes, you are correct.
And I understand why you are asking this :) 

I need to make changes so a sliding window attention is correctly applied to such models as Mistral that have SWA on every single layer.

Reply:
I made changes so now, depending on the config, SWA is applied either to every layer (like in Mistral) or to every other layer (like in Gemma).
The only problem that I have is the naming.
Maybe @rasbt can find something more suitable?

Reply:
you mean instead of "all" and "interleaved"? Perhaps "every" and "every-other" but to be honest, "all" and "interleaved" is not bad.

Reply:
no, I meant `swa_apply_to_layers` 🙂 
because than I do `not idx % self.swa_apply_to_layers`

Feels weird when you read the code

Reply:
Hm, I think it is fine but then I know exactly what this is because I saw all the other context. Maybe it will be confusing in a few months when we semi-forgot about it. Small change, but I wonder if this would be slightly more readable?

```python
        should_apply_swa = (
            config.sliding_window_size is not None and 
            block_idx % config.sliding_window_frequency == 0
        )
        self.apply_sliding_window_attention = should_apply_swa
```

(where `swa_apply_to_layers` -> `sliding_window_frequency`)

Reply:
yep yep, it's better.
Not `frequency`, we need something better. But the direction is right.

Reply:
`sliding_window_layer_interval`?

Reply:
The problem with interval is that with a value of 1 we apply to all layers, but 1 might mean that we apply to every other layer (1 layer skipping). Correct me if I'm wrong.

My vocabulary is really tiny, so I used an online tool to provide me a list of synonyms, so here we go:
1. sliding_window_layer_ordering
2. sliding_window_layer_arrangement
3. sliding_window_layer_setup
4. sliding_window_layer_positioning
5. sliding_window_layer_placing
6. sliding_window_layer_layout
7. sliding_window_layer_config

P.S. We are full in a bikeshedding territory now 😆.

Although, maybe an interval is not that bad 🤔

Reply:
Hm, yeah, I think `interval` or `placing` works. The other ones are a bit too ambiguous

Reply:
I settled on `placing`.
Fits better to `all` and `interleaved` options.
Thanks for the help 👍